### ISSUE TITLE
Large scale reconstruction

### ISSUE THREAD

**Issue by Sebo-the-tramp** [nerfstudio-project/nerfstudio#3617]
Thank you for your amazing work!

My question is about Gaussian Splatting (Splatfacto/Splatfacto-Big). What is the limit of images that can be processed in one training epoch? E.g. if I have 4000 images, will Splatfacto use all of them? Because when I activate the viewer, I only see a subset of them (maybe is done for efficiency reasons?). So at the end of the training, the result is sub-optimal.

My questions are then:
1) are all the images used or rather only a subset is used if the number of images is too high?
2) consequently are the results sub-optimal because not all images are used to fit the scene or because the training is only done for 30k steps (which might be not much for that many images)?
3) Is there a way to set the maximum number of images to be used in the scene as well the maximum number of Gaussians?

I know these are a lot of questions, but even addressing only one could help a lot!

Thank you so much :)

**Comment by Sebo-the-tramp** [comment]
Thank you for your reply!
My current training script is the following:

```bash
ns-train splatfacto-big  --data ./global_OG/sfm --output_dir ./splatfacto --pipeline.model.stop-split-at 400000 \ 
--max-num-iterations 500000 --steps-per-eval-all-images 50000 --machine.num-devices 1 \ 
--pipeline.model.warmup-length 5000 --pipeline.model.refine-every 1000 --pipeline.model.num-downscales 4 \
--pipeline.model.camera-optimizer.mode "SO3xR3" --viewer.quit-on-train-completion True
```

I don't use COLMAP in the training because I ran `ns-process-data` before, so it seems I cannot run your suggestion with COLMAP. After I run the above command, the results still seem suboptimal, with only 240k Gaussian being used at the end to represent the whole reconstruction. I wonder why this happens, what if I want to use 10 million gaussians? Is that just a limitation of the code to come up with the final number of Gaussians? 

Thank you so much!!

**Comment by jinwookpark** [comment]
![Image](https://github.com/user-attachments/assets/86597209-d78e-4603-b25f-4ea7f3c1d6f1)

Even if you didn't use COLMAP and configured SfM using another approach, you can still use the maximum number of poses for training in the following manner.

EX)
ns-train splatfacto-big  --data ./global_OG/sfm --output_dir ./splatfacto --pipeline.model.stop-split-at 400000 \ 
--max-num-iterations 500000 --steps-per-eval-all-images 50000 --machine.num-devices 1 \ 
--pipeline.model.warmup-length 5000 --pipeline.model.refine-every 1000 --pipeline.model.num-downscales 4 \
--pipeline.model.camera-optimizer.mode "SO3xR3" --viewer.quit-on-train-completion True \
nerfstudio-data --downscale-factor 4 --eval-mode fraction --train-split-fraction 1.0 


In my personal opinion, ns-train splatfacto-big does not allow you to specify the maximum number of Gaussians since it optimizes the number internally. However, the latest nerfstudio's ns-train splatfacto-mcmc includes an option

--pipeline.model.max-gs-num 1000000

Using this option, you may be able to utilize up to 10M Gaussians.

**Comment by tyl666tyl** [comment]
Hello! I confront the same problem with yours. Did you get a good result in a large scale scene? I have 5000 images from simulate environment and don't get a good result by using splatfacto.
Thank you so much :)
